### DISCUSSION TITLE
Trying to read undefined property view

### DISCUSSION THREAD

**Discussion by sandyHM** [pimcore/pimcore#12618] [Q&A / Support] [answered]
Trying to read undefined property view in pimcore bundle controller
![Screenshot from 2022-07-11 11-30-59](https://user-images.githubusercontent.com/107986459/178199675-e2866a71-59ab-4c04-bec3-db8f4ca99bf7.png)

**Comment by dvesh3** [comment, marked as answer]
Auto view rendering and ViewModel was removed in version 10.0.0. please see https://github.com/pimcore/pimcore/pull/7247.

Return `$this->render()` in your controller along with params.

**Reply by sandyHM** [reply]
`$this->render()` use for rendering template
but i need to know what is the role of `$this->view` inside controller

and when i tried to access `$this->view` in any Controller action it gave me
Trying to read undefined property view\

**Reply by jdreesen** [reply]
> and when i tried to access $this->view in any Controller action it gave me
Trying to read undefined property view\

That's because `$this->view` doesn't exist in Pimcore 10.

**Reply by sandyHM** [reply]
**Thank you** [dvesh3](https://github.com/dvesh3) and [jdreesen](https://github.com/jdreesen)

**Comment by sandyHM** [comment]
Thank you [dvesh3](https://github.com/dvesh3) and [jdreesen](https://github.com/jdreesen)
